### ISSUE TITLE
TypeScript: Cannot use Http2SecureServer

### ISSUE THREAD

**Issue by williamd5** [socketio/socket.io#4494]
**Describe the bug**
The documentation indicates the following code for server initialisation with http2:
```ts
import { createSecureServer } from "http2";
import { Server } from "socket.io";

const httpServer = createSecureServer({...});

const io = new Server(httpServer, { /* options */ });
```
`httpServer` is `http2.Http2SecureServer`
when calling `new Server(httpServer, {})`, `TS2769: No overload matches this call.` None of the signatures accepts Http2SecureServer.

The full error is:
```
TS2769: No overload matches this call.
   Overload 1 of 3, '(srv?: number | import("http").Server<...> | import("https").Server<...> | undefined, opts?: Partial<...> | undefined): Server<...>', gave the following error.
     Argument of type 'Http2SecureServer' is not assignable to parameter of type 'number | Server<...> | Server<...> | undefined'.
       Type 'Http2SecureServer' is missing the following properties from type 'Server<typeof IncomingMessage, typeof ServerResponse>': closeAllConnections, closeIdleConnections, maxHeadersCount, maxRequestsPerSocket, and 4 more.
   Overload 2 of 3, '(srv: number | import("http").Server<...> | import("https").Server<...> | Partial<...> | undefined, opts?: Partial<...> | undefined): Server<...>', gave the following error.     Argument of type 'Http2SecureServer' is not assignable to parameter of type 'number | Server<...> | Server<...> | Partial<...> | undefined'.
       Type 'Http2SecureServer' is not assignable to type 'Server<typeof IncomingMessage, typeof ServerResponse>'.
```

**To Reproduce**

Please fill the following code example:

Socket.IO server version: `4.5.2`

*Server*

(copy-pasted 1:1 from https://socket.io/docs/v4/server-initialization/#with-an-http2-server for TypeScript)
```ts
import { readFileSync } from "fs";
import { createSecureServer } from "http2";
import { Server } from "socket.io";

const httpServer = createSecureServer({
	allowHTTP1: true,
	key: readFileSync("/path/to/my/key.pem"),
	cert: readFileSync("/path/to/my/cert.pem")
});

const io = new Server(httpServer, { /* options */ });
```

**Platform:**
 - Device: Razer Book 13
 - OS: Ubuntu 22.04 LTS
 - Node.js: v16.17.1
 - npm: 8.15.0

**Comment by darrachequesne** [comment]
Hi! Thanks for the heads-up :+1: 

This should be fixed by https://github.com/socketio/socket.io/commit/d3d0a2d5beaff51fd145f810bcaf6914213f8a06 (awaiting release).

**Comment by williamd5** [comment]
Thank you. Looking forward to the release

**Comment by darrachequesne** [comment]
Included in [4.5.3](https://github.com/socketio/socket.io/releases/tag/4.5.3). Thanks for the heads-up :+1:
